FHEM/00_SIGNALduino.pm

```diff
@@ -1503,14 +1507,20 @@ sub SIGNALduino_SendFromQueue {
     #DevIo_SimpleWrite($hash, $msg . "\n", 2);
     $hash->{sendworking} = 1;
     SIGNALduino_SimpleWrite($hash,$msg);
-    if ($msg =~ m/^S[RCM];/) {
+    if ($msg =~ m/^S[RCMN];/) {
       	$hash->{ucCmd}->{cmd} = 'sendraw';
        	$hash->{ucCmd}->{timenow} = time();
        	$hash->{ucCmd}->{responseSub} = \&SIGNALduino_CheckSendRawResponse;
       	$hash->{logMethod}->($name, 4, "$name: SendFromQueue, msg=$msg"); # zu testen der Queue, kann wenn es funktioniert auskommentiert werden
     } elsif ($msg =~ "^e") {											# Werkseinstellungen
     	SIGNALduino_Get($hash,$name,'ccconf');
     	SIGNALduino_Get($hash,$name,'ccpatable'); 
+
+			## set rfmode to default from uC
+    	my $rfmode = AttrVal($name, 'rfmode', undef);
+    	#CommandAttr($hash,"$name rfmode SlowRF") if (defined $rfmode && $rfmode ne 'SlowRF');  # option with save question mark
+    	$attr{$name}{rfmode} = 'SlowRF' if (defined $rfmode && $rfmode ne 'SlowRF');            # option without save question mark
```

Comment:
Ich habe mich an die Tests gemacht, aber wieso setzen wir hier das attribut auf SlowRF wenn es nicht SlowRF ist.
Wir sollten dem Anwender nicht sein Attribut ändern, welchen Grund gibt es denn hierfür?

Reply:
Moin Moin.
Vorab, ja die "automatische" Anpassunng von Attributen ist teilweise hart um diskutiert.

>Ich habe mich an die Tests gemacht, aber wieso setzen wir hier das attribut auf SlowRF wenn es nicht SlowRF ist.

Unsere Defaulteinstellungen im uC entsprechen dem SlowRF Modus.

>Wir sollten dem Anwender nicht sein Attribut ändern, welchen Grund gibt es denn hierfür?

Der Grund ist folgendes.
Wenn ich in einem anderen Empfangsmodus bin und der Benutzer den Befehl `raw e` ausführt weil er von neu anfangen möchte mit den uC default Registerwerten (Bsp. zum Test des normalen ASK/OOK Empfanges) so bleibt das Attribut stur auf Bsp. Lacrosse_mode1 obwohl ASK/OOK definiert ist. Damit für den Benutzer keine wiedesprüchliche Fragen aufkommen und er das Attribut wieder setzen kann, so wurde diese Anpassung vorgenommen. (Das Attribut wird nur gesetzt wenn der alte, zu neuen Wert sich ändert)

Reply:
Anstelle das Attribut zu überschreiben könnten wir auch eine Warnung ausgeben.

Reply:
Meinst du, nur eine Warnung ausgeben und keine Werte vom Register zurücksetzen? Die Register müssen auf jedenfall wieder gesetzt werden.

Wenn wir anstelle der Warnung oder der "stillen" ÜberschreibMethode des Attributes die Variante nehmen das Attribut zu setzen mir dem geändert "?", so sieht der User auch, das sich was geändert hat. Dieser Fall birgt nur die Gefahr, das bei einem nicht speichern eine anderer Wert im Attribut steht als im Register.

Für den normalen Benutzer denke ich, wäre es am sichersten und verständlichsten wenn das Attribut aus SlowRF gesetzt wird ohne ? Der Hinweis ist denke ich "überflüssig" weil bei jedem anderen Wert des Attributes erfolgt auch keine Ausgabe. Ein Benutzer ändert doch das Attribut bewusst weil ich etwas ändern möchte.

Reply:
Eine Optimale Lösung gibt es da nicht.
Es kann halt Fragen aufwerfen,

a) wieso wurde mein Attribut geändert
b) Was ist mit Speichern, es ist ja nicht gespeichert bis ein save Befehl ausgeführt wird ohne das "?" weiss es der Anwender auch noch nicht mal dass er speicher sollte
c) Warum wurde ich nicht informiert, dass das Attribut geändert wurde.

Reply:
Ich denke, das Verhalten mal zu ergründen wie es genau beim CUL unsw ist. Da erfolgt auch keine Ausgabe und co. Da sollte man sich dies als Vorlage nehmen und nicht zu viel Möglichkeiten erörtern.

Reply:
Ich habe heute noch einmal nachgesehen und getestet.

Ich hätte hier eine Lösung wie wir das ganze angehen könnten. Der CUL Bsp setzt das Attribut mit dem Hinweis das man speichern muss. (normales Verhalten). Wenn wir nun beim Start das Register setzen nach dem Attributwert, so haben wir das richtige Verhalten wie es für den normalen Benutzer korrekt ist.